solution/index.js

```diff
@@ -0,0 +1,317 @@
+async function load (){
+    try {
+        const spin = document.getElementById("load");                    // add loading
+        spin.setAttribute("class","loader")
+        const url = "https://json-bins.herokuapp.com/bin/614adb274021ac0e6c080c13";
+        const headers = {  
+            Accept: "application/json",
+            "Content-Type": "application/json" 
+        }
+        const response = await fetch (url,{
+            method:"GET",
+            headers
+        })
+        const mydata = await response.json();
+        clear();                                                          // clear the page                    
+        localStorage.setItem("tasks", JSON.stringify(mydata.tasks.data)); // update local
+        data = mydata.tasks.data;                                         // update data
+        printData();                                                      // print the page 
+        spin.removeAttribute("class","loader");                           // remove loading
+    }
+    catch(error){
+        return alert("error")
+    }
+}
+
+// save the page
+document.getElementById("save-btn").addEventListener("click",save)
+async function save (){
+   try {
+        const spin = document.getElementById("load");                   // add loading
+        spin.setAttribute("class","loader")
+    
+        const url = "https://json-bins.herokuapp.com/bin/614adb274021ac0e6c080c13";
+        const body = JSON.stringify({"tasks":{data}})
+        const headers = {  
+            Accept: "application/json",
+            "Content-Type": "application/json" 
+        }
+        const response = await fetch (url,{
+            method: "PUT",
+            body,
+            headers
+        }) 
+        spin.removeAttribute("class","loader");                         // remove loading
+    }
+    catch(error){
+        return alert("error")
+    }
+}
+
+// clear the tasks in the page 
+document.getElementById("clear-btn").addEventListener("click",clear)
+function clear (){
+    const tasks = document.querySelectorAll(".task")
+    for(const task of tasks){
+        task.remove();
+    }
+    data = {
+        "todo": [],
+        "in-progress": [],
+        "done": []
+    };
+    localStorage.setItem("tasks",JSON.stringify(data));
+}
+
+// create an element 
+function createElement(tagName, text , classes = [], attributes = {}, eventListeners = {}){
+
+    const element = document.createElement(tagName);
+
+    // text in the element
+    element.textContent = text;
+
+    // Classes
+    for(const cls of classes) {
+      element.classList.add(cls);
+    }
+    // Attributes
+    for (const attr in attributes) {
+      element.setAttribute(attr, attributes[attr]);
+    }
+    // Event
+    for (act in eventListeners) {
+      element.addEventListener(act, eventListeners[act]);
+    }
+    return element;
+}
+
+// print the data
+printData();
+function printData(){
```

Comment:
is this function for printing the data? your adding events in it.

Reply:
Yes this function prints to page all the "tasks" stored in the data (localstorage)